### PR TITLE
Fix lint errors for public classes without docs

### PR DESCRIPTION
**Pull request by gugalo**
**Jira issue:**
https://glia.atlassian.net/browse/MOB-3172

**What was solved?**

- 54 classes got @hide mark for docs
- 28 classes converted from public Java to Kotlin internal
- 5 Kotlin classes had wrong access modifier
- New documentation added to 6 classes
- A lot of deprecated classes got simple in-code note

These are lint fixes that are required for CI checks in [PR#976](https://github.com/salemove/android-sdk-widgets/pull/976) to pass.

Sorry, I did not think that there would be so many changes and have not taken into account that these changes should be separated into smaller PRs for easier review when started working on this.

**Release notes:**

 - [x] Feature: some classes got additional docs. Hiden some classes from the public
 - [ ] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Tests updated? Added? Unit, acceptance, snapshots?
 - [ ] Logging for future troubleshooting of client issues added?

**Screenshots:**

### CODE REVIEW

**Review by DavDo — APPROVED** [review]

**Review by AndriiHorishniiMOC — APPROVED** [review]
LGTM!

File: widgetssdk/src/main/java/com/glia/widgets/StringKeyPair.kt [hunk 1]
```diff
@@ -1,7 +1,13 @@
 package com.glia.widgets
 
+/**
+ * Used for retrieving custom locales strings that should have placeholders replaced with real values placeholders
+ */
 data class StringKeyPair(val key: StringKey, val value: String)
 
+/**
+ * List placeholders for [StringKeyPair]
```
Comment (kadriiiii) [inline widgetssdk/src/main/java/com/glia/widgets/StringKeyPair.kt:9]:
```suggestion
 * List placeholders for [StringKeyPair].
```

File: widgetssdk/src/main/java/com/glia/widgets/StringProvider.java [hunk 2]
```diff
@@ -3,6 +3,11 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.StringRes;
 
+/**
+ * Class used for retrieving string from custom locales
```
Comment (kadriiiii) [inline widgetssdk/src/main/java/com/glia/widgets/StringProvider.java:7]:
```suggestion
 * Retrieving string from custom locales.
```

File: widgetssdk/src/main/java/com/glia/widgets/UiTheme.kt [hunk 3]
```diff
@@ -19,6 +19,12 @@ import com.glia.widgets.view.unifiedui.theme.base.ColorTheme
 import com.glia.widgets.view.unifiedui.theme.defaulttheme.AlertTheme
 import kotlinx.parcelize.Parcelize
 
+/**
+ * Class used for customising Glia UI at runtime
```
Comment (kadriiiii) [inline widgetssdk/src/main/java/com/glia/widgets/UiTheme.kt:23]:
```suggestion
 * Customize Glia UI at runtime.
```

File: widgetssdk/src/main/java/com/glia/widgets/UiTheme.kt [hunk 4]
```diff
@@ -458,6 +464,12 @@ data class UiTheme(
         return AlertDialogConfiguration(theme, properties, icons)
     }
 
+    /**
+     * Builder for Glia UI customisation at runtime
```
Comment (kadriiiii) [inline widgetssdk/src/main/java/com/glia/widgets/UiTheme.kt:468]:
```suggestion
     * Builder for Glia UI customization at runtime.
```

File: widgetssdk/src/main/java/com/glia/widgets/core/visitor/GliaWidgetException.java [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 import com.glia.androidsdk.GliaException;
 
+/**
+ * Glia Exception that has been thrown by Widgets SDK
```
Comment (kadriiiii) [inline widgetssdk/src/main/java/com/glia/widgets/core/visitor/GliaWidgetException.java:6]:
```suggestion
 * Glia Exception that has been thrown by Widgets SDK.
```

**Review by kadriiiii — COMMENTED** [review]
Suggested a few rewordings :)

**Comment by gugalo** [comment]
All suggestions and Unit test should be fixed

**Review by kadriiiii — APPROVED** [review]
Thank you for updates!

**Comment by gugalo** [comment]
!squash

**Pull request merged**